### ISSUE TITLE
Assigning defined variable on object destructuring errors with "Bad Assignment"

### ISSUE THREAD

**Issue by DelvarWorld** [eslint/eslint#2424]
This works:

    var {error} = data;

This errors but it's likely supposed to error (?)

    var error;
    {error} = data; // Missing semicolon.

This errors but likely is **not** supposed to error (?)

    var error;
    ({error} = data); // Bad assignment

**Comment by nzakas** [comment]
We need some more information:

1. Please paste the actual eslint output for each example.
1. What version of ESLint are you using?
1. Are you using babel-eslint or the default parser?
1. What does your config look like?

**Comment by gyandeeps** [comment]
@DelvarWorld we haven't received any info. We will close this if we don't get any response.

**Comment by xjamundx** [comment]
I can't reproduce this with eslint `v0.21.0`. 

Here is my `test.js` file:

```js
let error, data = {};
({error} = data);
console.log(error);
```

`eslint --no-eslintrc --reset --env es6 test.hjs` gives no errors.

Also espree handles this just fine:

```js
{ type: 'Program',
  body: 
   [ { type: 'ExpressionStatement',
       expression: 
        { type: 'AssignmentExpression',
          operator: '=',
          left: 
           { type: 'ObjectPattern',
             properties: 
              [ { type: 'Property',
                  key: { type: 'Identifier', name: 'error' },
                  value: { type: 'Identifier', name: 'error' },
                  kind: 'init',
                  method: false,
                  shorthand: true,
                  computed: false } ] },
          right: { type: 'Identifier', name: 'data' } } } ],
  sourceType: 'script' }
```

Pretty sure this is not an issue with eslint.

**Comment by ilyavolodin** [comment]
This doesn't look like an ESLint issue. Feel free to reopen with more information if it's still happening.
